File: Python/student_management_system/details_generator.py
import random
from utils import COMBINED_SCIENCES, PURE_SCIENCES, HUMANITIES, COURSEWORK_NON_COURSEWORK


def generate_random_marks():
    subjects = ["English", "Mother Tongue", "Maths"]
    subjects.extend(generate_science_subjects())
    subjects.extend(generate_humanities_subjects())
    subjects.extend(generate_coursework_non_coursework_subjects())

    return {subject: random.randint(60, 100) for subject in subjects}


def generate_science_subjects():
    science_type = random.choice(["combined", "pure"])
    if science_type == "combined":
        return [random.choice(COMBINED_SCIENCES)]
    else:
        return random.sample(PURE_SCIENCES, 2)


def generate_humanities_subjects():
    return [random.choice(HUMANITIES)]


def generate_coursework_non_coursework_subjects():
    num_courses = random.choice([1, 2])
    return random.sample(COURSEWORK_NON_COURSEWORK, num_courses)


def generate_random_phone_number():
    return f"{random.randint(6000, 9999)}{random.randint(1000, 9999)}"

File: Javascript/student_management_system/detailsGenerator.js
import {
    COMBINED_SCIENCES,
    PURE_SCIENCES,
    HUMANITIES,
    COURSEWORK_NON_COURSEWORK,
} from "./utils.js";

function getRandomInt(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

export function generateRandomMarks() {
    const subjects = ["English", "Mother Tongue", "Maths"];
    subjects.push(...generateScienceSubjects());
    subjects.push(...generateHumanitiesSubjects());
    subjects.push(...generateCourseworkNonCourseworkSubjects());

    const marks = {};
    subjects.forEach((subject) => {
        marks[subject] = getRandomInt(60, 100);
    });

    return marks;
}

function generateScienceSubjects() {
    const scienceType = Math.random() < 0.5 ? "combined" : "pure";
    if (scienceType === "combined") {
        return [
            COMBINED_SCIENCES[
                Math.floor(Math.random() * COMBINED_SCIENCES.length)
            ],
        ];
    } else {
        const selected = [];
        while (selected.length < 2) {
            const subject =
                PURE_SCIENCES[Math.floor(Math.random() * PURE_SCIENCES.length)];
            if (!selected.includes(subject)) {
                selected.push(subject);
            }
        }
        return selected;
    }
}

function generateHumanitiesSubjects() {
    return [HUMANITIES[Math.floor(Math.random() * HUMANITIES.length)]];
}

function generateCourseworkNonCourseworkSubjects() {
    const numCourses = Math.random() < 0.5 ? 1 : 2;
    const shuffled = COURSEWORK_NON_COURSEWORK.sort(() => 0.5 - Math.random());
    return shuffled.slice(0, numCourses);
}

export function generateRandomPhoneNumber() {
    return `${getRandomInt(6000, 9999)}${getRandomInt(1000, 9999)}`;
}
